packages/react/src/Pillar/Pillar.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {forwardRef, PropsWithChildren, HTMLAttributes, type Ref} from 'react'
+import clsx from 'clsx'
+import {Heading, HeadingTags, HeadingProps, Text} from '..'
+import type {BaseProps} from '../component-helpers'
+import {Colors} from '../constants'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/pillar/colors-with-modes.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './Pillar.module.css'
+import {Icon as IconProps} from '@primer/octicons-react'
+
+export const PillarIconColors = Colors
+
+export const defaultPillarIconColor = PillarIconColors[0]
+export type PillarProps<C extends keyof JSX.IntrinsicElements = 'div'> = React.HTMLAttributes<C> & {
+  /**
+   * The HTML element used to render the grid.
+   */
+  as?: C | 'div' | 'span' | 'article'
+  /**
+   * Valid children include Pillar.Image, Pillar.Heading, and Pillar.Description
+   */
+  children:
+    | React.ReactNode
+    | React.ReactElement<PillarIconProps>
+    | React.ReactElement<PillarHeadingProps>
+    | React.ReactElement<PillarDescriptionProps>
```

Comment:
Wonder if we could just use PropsWithChildren? I think declaring React.ReactNode will make the other types redundant.